### ISSUE TITLE
systems.listRolesForUser throws error

### ISSUE THREAD

**Issue by jturcino** [TACC/agavepy#19]
When trying to use .systems.listRolesForUser(), I get a TypeError. Here's what my script does:
```
#!/usr/bin/env python
from agavepy.agave import Agave
import os.path
import json

if os.path.isfile(os.path.expanduser('~/.vdjapi')) is True:
        with open(os.path.expanduser('~/.vdjapi'), 'r') as json_file:
            json_dict = json.load(json_file)
        json_file.close()
        access_token = str(json_dict['access_token'])

my_agave = Agave(api_server = 'https://vdj-agave-api.tacc.utexas.edu', token = access_token)
roles_list = my_agave.systems.listRolesForUser(systemId = 'jturcino-stampede-storage', username = 'jturcino')

print 'Roles are:', roles_list
```
When I run this, I get this error:
```
$ ./trial-file.py 
Traceback (most recent call last):
  File "./tester.py", line 14, in <module>
    roles_list = my_agave.systems.listRolesForUser(systemId = 'jturcino-stampede-storage', username = 'jturcino')
  File "/home1/03761/jturcino/.local/lib/python2.7/site-packages/agavepy/agave.py", line 419, in __call__
    processed = self.post_process(resp.json(), self.return_type)
  File "/home1/03761/jturcino/.local/lib/python2.7/site-packages/agavepy/agave.py", line 442, in post_process
    return self.process_model(obj, return_type)
  File "/home1/03761/jturcino/.local/lib/python2.7/site-packages/agavepy/agave.py", line 473, in process_model
    for k in obj})
  File "/home1/03761/jturcino/.local/lib/python2.7/site-packages/agavepy/agave.py", line 473, in <dictcomp>
    for k in obj})
  File "/home1/03761/jturcino/.local/lib/python2.7/site-packages/agavepy/agave.py", line 441, in post_process
    return f(obj, return_type)
  File "/home1/03761/jturcino/.local/lib/python2.7/site-packages/agavepy/agave.py", line 451, in process_array
    for elem in obj]
  File "/home1/03761/jturcino/.local/lib/python2.7/site-packages/agavepy/agave.py", line 442, in post_process
    return self.process_model(obj, return_type)
  File "/home1/03761/jturcino/.local/lib/python2.7/site-packages/agavepy/agave.py", line 473, in process_model
    for k in obj})
  File "/home1/03761/jturcino/.local/lib/python2.7/site-packages/agavepy/agave.py", line 473, in <dictcomp>
    for k in obj})
TypeError: string indices must be integers
```

**Comment by joestubbs** [comment]
Thanks for the bug report. I am going to try and get to this today, should be relatively quick, but I have two other pressing things on my plate that need to get done before 3/1.

**Comment by joestubbs** [comment]
This is related to a bug in Agave. When you query the system/roles/<username> endpoint it does not properly send a list of roles. For example:

```
$ curl -H "Authorization: Bearer $tok" $base/systems/v2/python-sdk-test-stor-jfs/roles/fooy

{
    "message": null,
    "result": {
        "_links": {
            "parent": {
                "href": "https://public.tenants.prod.agaveapi.co/systems/v2/python-sdk-test-stor-jfs"
            },
            "profile": {
                "href": "https://public.tenants.prod.agaveapi.co/profiles/v2/fooy"
            },
            "self": {
                "href": "https://public.tenants.prod.agaveapi.co/systems/v2/python-sdk-test-stor-jfs/roles/fooy"
            }
        },
        "role": "OWNER",
        "username": "fooy"
    },
    "status": "success",
    "version": "2.1.6-rbf84964"
}
```
Note that result is a dictionary. However, if you query the system roles endpoint instead you do get a list:

```
 $ curl -H "Authorization: Bearer $tok" $base/systems/v2/python-sdk-test-stor-jfs/roles
{
    "message": null,
    "result": [
        {
            "_links": {
                "parent": {
                    "href": "https://public.tenants.prod.agaveapi.co/systems/v2/python-sdk-test-stor-jfs"
                },
                "profile": {
                    "href": "https://public.tenants.prod.agaveapi.co/profiles/v2/fooy"
                },
                "self": {
                    "href": "https://public.tenants.prod.agaveapi.co/systems/v2/python-sdk-test-stor-jfs/roles/fooy"
                }
            },
            "role": "OWNER",
            "username": "fooy"
        }
    ],
    "status": "success",
    "version": "2.1.6-rbf84964"
}
```

I'll get a ticket raised in the Agave Jira system.

**Comment by joestubbs** [comment]
According to Rion, a user can have at most one role on a system. Therefore, I have updated the swagger definition file appropriately. In particular, the method name is now "getRoleForUser", for example:

```
In [3]: role = ag2.systems.getRoleForUser(systemId='python-sdk-test-stor-jfs', username='fooy')
In [4]: role
Out[4]: 
{u'_links': {u'parent': {u'href': u'https://public.tenants.prod.agaveapi.co/systems/v2/python-sdk-test-stor-jfs'},
  u'profile': {u'href': u'https://public.tenants.prod.agaveapi.co/profiles/v2/fooy'},
  u'self': {u'href': u'https://public.tenants.prod.agaveapi.co/systems/v2/python-sdk-test-stor-jfs/roles/fooy'}},
 u'role': u'OWNER',
 u'username': u'fooy'}
In [5]: role.role
Out[5]: u'OWNER'
```

This fix is available in the v0.4.0 branch.
